### PR TITLE
Fix ineffective breaks

### PR DESCRIPTION
**Pull request by adlerjohn**
Follow-on of #260

### CODE REVIEW

File: state/indexer/block/kv/kv.go [hunk 1]
```diff
@@ -199,10 +201,13 @@ func (idx *BlockerIndexer) Search(ctx context.Context, q *query.Query) ([]int64,
 
 		select {
 		case <-ctx.Done():
-			break
+			cont = false
```
Comment (evan-forbes) [inline state/indexer/block/kv/kv.go:204]: wait, I'm missing something here. What does breaking `if !cont` change as opposed to simply breaking here?

Comment (adlerjohn) [reply state/indexer/block/kv/kv.go:204]: `break` in the original code will break out of the `select` (i.e. do literally nothing), instead of the `for`.

**Review by tzdybal — APPROVED** [review]
This is in the code "ported" (aka copy-pasted) from Tendermint. This functionality should be removed soon™ from Tendermint and Optimint.

But let's fix this.

**Comment by adlerjohn** [comment]
> This is in the code "ported" (aka copy-pasted) from Tendermint. This functionality should be removed soon™ from Tendermint and Optimint.

Uhhh 😂 is this something that upstream should be made aware of?

**Comment by liamsi** [comment]
> Uhhh 😂 is this something that upstream should be made aware of?

I'm pretty sure upstream looks much cleaner and very different now already. Actually the solution there is: https://github.com/tendermint/tendermint/blob/fbe86ea6459ec96f5bf427789c8109247ac6b302/internal/state/indexer/block/kv/kv.go#L187-L201

**Pull request merged**